### PR TITLE
Removed EntityManager member variable from Components and EntitySystems

### LINKED ISSUE #2485: Remove EntityManager member variable from various things
Both entities and EntitySystems have a property with this already so any components / systems that store it should be updated.

### PR DESCRIPTION
**Pull request by ColdAutumnRain**
Fixes #2485

### CODE REVIEW

File: Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs [hunk 1]
```diff
@@ -33,7 +33,6 @@ public class TraitorOverlay : Overlay
             IEyeManager eyeManager)
             : base(nameof(TraitorOverlay))
         {
-            _entityManager = entityManager;
```
Comment (Zumorica) [inline Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs:36]: Why did you remove this? The game will crash when the overlay is applied.

File: Content.Server/GameObjects/EntitySystems/Atmos/AtmosDebugOverlaySystem.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public sealed class AtmosDebugOverlaySystem : SharedAtmosDebugOverlaySystem
     {
         [Robust.Shared.IoC.Dependency] private readonly IGameTiming _gameTiming = default!;
         [Robust.Shared.IoC.Dependency] private readonly IPlayerManager _playerManager = default!;
-        [Robust.Shared.IoC.Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/Atmos/AtmosDebugOverlaySystem.cs:28]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/AtmosExposedSystem.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace Content.Server.GameObjects.EntitySystems
     public class AtmosExposedSystem
     : EntitySystem
     {
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/AtmosExposedSystem.cs:14]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/Click/ExamineSystem.cs [hunk 4]
```diff
@@ -11,7 +11,7 @@ namespace Content.Server.GameObjects.EntitySystems.Click
 {
     public class ExamineSystem : ExamineSystemShared
     {
-        [Dependency] private IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/Click/ExamineSystem.cs:14]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 5]
```diff
@@ -41,7 +41,7 @@ namespace Content.Server.GameObjects.EntitySystems.Click
     public sealed class InteractionSystem : SharedInteractionSystem
     {
         [Dependency] private readonly IMapManager _mapManager = default!;
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:44]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/ConstructionSystem.cs [hunk 6]
```diff
@@ -34,7 +34,7 @@ namespace Content.Server.GameObjects.EntitySystems
     [UsedImplicitly]
     internal class ConstructionSystem : SharedConstructionSystem
     {
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/ConstructionSystem.cs:37]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/HandsSystem.cs [hunk 7]
```diff
@@ -27,7 +27,7 @@ namespace Content.Server.GameObjects.EntitySystems
     [UsedImplicitly]
     internal sealed class HandsSystem : EntitySystem
     {
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/HandsSystem.cs:30]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/MoverSystem.cs [hunk 8]
```diff
@@ -35,7 +35,7 @@ internal class MoverSystem : SharedMoverSystem
         [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager = default!;
         [Dependency] private readonly IMapManager _mapManager = default!;
         [Dependency] private readonly IRobustRandom _robustRandom = default!;
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/MoverSystem.cs:38]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/VerbSystem.cs [hunk 9]
```diff
@@ -17,7 +17,7 @@ namespace Content.Server.GameObjects.EntitySystems
 {
     public class VerbSystem : SharedVerbSystem, IResettingEntitySystem
     {
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/VerbSystem.cs:20]:
```suggestion
```

**Review by Zumorica — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

File: Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs (hunk 1 above)
Comment (ColdAutumnRain) [reply Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs:36]: Sorry, my bad. I used a script that removes lines with `_entityManager` to delete all `private readonly IEntityManager _entityManager = default!;` lines. But I think that some other lines was affected as well.

File: Content.Client/GameObjects/Components/Suspicion/SuspicionRoleComponent.cs [hunk 10]
```diff
@@ -21,7 +21,7 @@ namespace Content.Client.GameObjects.Components.Suspicion
     public class SuspicionRoleComponent : SharedSuspicionRoleComponent
     {
         [Dependency] private readonly IGameHud _gameHud = default!;
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+
```
Comment (Clyybber) [inline Content.Client/GameObjects/Components/Suspicion/SuspicionRoleComponent.cs:24]:
```suggestion
```

File: Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs [hunk 11]
```diff
@@ -33,6 +33,7 @@ public class TraitorOverlay : Overlay
             IEyeManager eyeManager)
             : base(nameof(TraitorOverlay))
         {
+
```
Comment (Clyybber) [inline Content.Client/GameObjects/Components/Suspicion/TraitorOverlay.cs:36]:
```suggestion
```

File: Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs [hunk 12]
```diff
@@ -387,6 +387,7 @@ public bool ToggleBuckle(IEntity user, IEntity to, bool force = false)
             return TryBuckle(user, to);
         }
 
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs:390]:
```suggestion
```

File: Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs [hunk 13]
```diff
@@ -410,6 +411,7 @@ public override void OnRemove()
         {
             base.OnRemove();
 
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs:414]:
```suggestion
```

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 14]
```diff
@@ -38,6 +38,7 @@ namespace Content.Server.GameObjects.Components.PDA
     public class PDAComponent : SharedPDAComponent, IInteractUsing, IActivate, IUse, IAccess
     {
         [Dependency] private readonly IPDAUplinkManager _uplinkManager = default!;
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/PDA/PDAComponent.cs:41]:
```suggestion
```

File: Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs [hunk 15]
```diff
@@ -29,7 +29,9 @@ namespace Content.Server.GameObjects.Components.Singularity
     [RegisterComponent]
     public class SingularityComponent : Component, ICollideBehavior
     {
+
         [Dependency] private IEntityManager _entityManager = null!;
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs:34]:
```suggestion
        [Dependency] private IEntityManager _entityManager = null!;
```

File: Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs [hunk 16]
```diff
@@ -29,7 +29,9 @@ namespace Content.Server.GameObjects.Components.Singularity
     [RegisterComponent]
     public class SingularityComponent : Component, ICollideBehavior
     {
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs:32]:
```suggestion
```

File: Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs (hunk 15 above)
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Singularity/SingularityComponent.cs:34]:
```suggestion
```

File: Content.Server/GameObjects/Components/Stack/StackComponent.cs [hunk 17]
```diff
@@ -24,9 +24,7 @@ namespace Content.Server.GameObjects.Components.Stack
     [ComponentReference(typeof(SharedStackComponent))]
     public class StackComponent : SharedStackComponent, IInteractUsing, IExamine
     {
-        [Dependency] private IEntityManager _entityManager = default!;
-
-        private bool _throwIndividually = false;
+         private bool _throwIndividually = false;
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Stack/StackComponent.cs:27]:
```suggestion
        private bool _throwIndividually = false;
```

File: Content.Server/GameObjects/EntitySystems/AI/Pathfinding/PathfindingSystem.cs [hunk 18]
```diff
@@ -337,6 +337,7 @@ private void HandleEntityMove(MoveEvent moveEvent)
 
             // The pathfinding graph is tile-based so first we'll check if they're on a different tile and if we need to update.
             // If you get entities bigger than 1 tile wide you'll need some other system so god help you.
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/EntitySystems/AI/Pathfinding/PathfindingSystem.cs:340]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/Click/ExamineSystem.cs [hunk 19]
```diff
@@ -21,6 +19,7 @@ static ExamineSystem()
             _entityNotFoundMessage.AddText(Loc.GetString("That entity doesn't exist"));
         }
 
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/EntitySystems/Click/ExamineSystem.cs:22]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 20]
```diff
@@ -152,7 +152,9 @@ private void InteractionActivate(IEntity user, IEntity used)
         private bool HandleWideAttack(ICommonSession session, EntityCoordinates coords, EntityUid uid)
         {
             // client sanitization
+
             if (!coords.IsValid(_entityManager))
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:157]:
```suggestion
```

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 21]
```diff
@@ -152,7 +152,9 @@ private void InteractionActivate(IEntity user, IEntity used)
         private bool HandleWideAttack(ICommonSession session, EntityCoordinates coords, EntityUid uid)
         {
             // client sanitization
+
```
Comment (Clyybber) [inline Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:155]:
```suggestion
```

**Review by Clyybber — APPROVED** [review]

**Review by DrSmugleaf — APPROVED** [review]
